### PR TITLE
Create HA node group and policy based on HANodeConfig policy in the configmap

### PR DESCRIPTION
**Pull request by ading1977**
This pull request creates HA node groups and anti-affinity policy for those groups based on the `HANodeConfig` policy specified in the `kubeturbo` configmap, for example:

```
apiVersion: v1
data:
  turbo.config: |-
    {
...
        "HANodeConfig": {
           "nodeRoles": [ "master", "worker" ]
        },
...
    }
kind: ConfigMap

```

### CODE REVIEW

File: pkg/discovery/detectors/detector_config_test.go [hunk 1]
```diff
@@ -230,7 +232,7 @@ var masterNodeTests = []MasterNodeTestCase{
 func TestDetectorConfig_detectMasterNode(t *testing.T) {
 	for scenarioIndex, scenarioName := range []string{"validFullConfig", "missingMasterNodeDetectors",
 		"missingDaemonPodDetectors", "missingBoth", "missingList", "emptySection"} {
-		setupTest(t, scenarioName)
+		_, _ = setupTest(scenarioName)
```
Comment (chlam4) [inline pkg/discovery/detectors/detector_config_test.go]: Is the `_, _` necessary here?

File: pkg/discovery/detectors/detector_config_test.go [hunk 2]
```diff
@@ -262,7 +264,7 @@ var daemonPodTests = []DaemonPodTestCase{
 func TestDetectorConfig_detectDaemon(t *testing.T) {
 	for scenarioIndex, scenarioName := range []string{"validFullConfig", "missingMasterNodeDetectors",
 		"missingDaemonPodDetectors", "missingBoth", "missingList", "emptySection"} {
-		setupTest(t, scenarioName)
+		_, _ = setupTest(scenarioName)
```
Comment (chlam4) [inline pkg/discovery/detectors/detector_config_test.go]: Same question as above.

File: pkg/discovery/detectors/detector_config_test.go (hunk 1 above)
Comment (ading1977) [reply pkg/discovery/detectors/detector_config_test.go]: This is not necessary but nice to have IMO for the following reasons:
* Get rid of the unhandled error code inspection warning in IntelliJ IDE
* Communicate to the code reader that we are deliberately ignoring a potential error

File: pkg/discovery/detectors/detector_config_test.go (hunk 2 above)
Comment (ading1977) [reply pkg/discovery/detectors/detector_config_test.go]: Same as above.

**Pull request merged**